core/src/services/gcs/core.rs

```diff
@@ -116,6 +131,15 @@ impl GcsCore {
     }
 
     pub async fn sign<T>(&self, req: &mut Request<T>) -> Result<()> {
+        if let Some(token) = &self.token {
```

Comment:
We don't need to change the logic here since `load_token` has already handled this.

Reply:
Makes sense :+1: 


Do I understand rightly that the same is also true for `sign_query`?

https://github.com/apache/opendal/blob/55fc7629a10fb4a17dcd7a78e2d3299efa8182b1/core/src/services/gcs/core.rs#L163-L170

I've added :point_up:, but the `sign_query` is handled through `load_credential` and _not_ `load_token`